app-ios/tutanota/Sources/Alarms/RepeatRule.swift

```diff
@@ -31,19 +31,45 @@ enum RepeatEndCondition : Equatable {
   }
 }
 
+struct EncryptedDateWrapper: Codable, Hashable {
+  let date: Base64
+}
+
 struct EncryptedRepeatRule : Codable, Hashable {
   let frequency: Base64
   let interval: Base64
   let timeZone: Base64
   let endType: Base64
   let endValue: Base64?
+  let excludedDates: [EncryptedDateWrapper]
+  
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    if let excludedDates = try container.decodeIfPresent([EncryptedDateWrapper].self, forKey: .excludedDates) {
+      self.excludedDates = excludedDates
+    } else {
+      self.excludedDates = [EncryptedDateWrapper]()
+    }
+    
+    self.frequency = try! container.decode(Base64.self, forKey: .frequency)
```

Comment:
why are those hard unwrapped?

Reply:
there's no reasonable default if they're missing

Reply:
I've meant that it uses `try!` instead of `try`

Reply:
right, fixed